### PR TITLE
remove io from tests

### PR DESCRIPTION
**Pull request by KuabeM**

### CODE REVIEW

File: src/intelhex.rs [hunk 1]
```diff
@@ -100,9 +100,17 @@ mod test {
 
     #[test]
     fn test_hex_convert() {
-        let file = PathBuf::from("tmp/test.hex");
+        let file = ":1000000001000000000000000200000000000000ED
```
Comment (drahnr) [inline src/intelhex.rs:103]: should this be prefixed by a `b` such that it is not null terminated `b""`

File: src/intelhex.rs [hunk 2]
```diff
@@ -131,7 +139,14 @@ mod test {
 
     #[test]
     fn test_write_hex() {
-        let name = PathBuf::from("tmp/test_write.hex");
+        let name = PathBuf::from("tmp_write.hex");
+        let hex = ":1000000001000000000000000200000000000000ED
```
Comment (drahnr) [inline src/intelhex.rs:144]: cc

File: src/intelhex.rs [hunk 3]
```diff
@@ -151,7 +166,7 @@ mod test {
 
         let mut hex_file = OpenOptions::new()
             .read(true)
-            .open("tmp/test_write.hex")
+            .open(name.clone())
```
Comment (drahnr) [inline src/intelhex.rs]: `.as_str()` should be fine

File: src/intelhex.rs [hunk 4]
```diff
@@ -161,23 +176,21 @@ mod test {
...
         assert_eq!(content, hex);
+
+        // delete file
+        std::fs::remove_file(name).expect("failed to delete tmp file");
     }
 
     #[test]
     fn bad_records() {
-        let bad_hex = PathBuf::from("tmp/bad_records.hex");
+        let bad_hex = ":10000000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFED
```
Comment (drahnr) [inline src/intelhex.rs:190]: cc

**Review by drahnr — CHANGES_REQUESTED** [review]

File: src/intelhex.rs (hunk 1 above)
Comment (KuabeM) [reply src/intelhex.rs:103]: A `String` is not null terminated, so I used that one. A byte string literal can be a non-utf8 string, i.e. the conversion might fail. (I hope I got that right)

**Review by drahnr — APPROVED** [review]

**Pull request merged**